src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWP.cs

```diff
@@ -376,6 +382,9 @@ private async void Launch(IPublicAPI api)
                 });
             }
 
+            // Empty constructor, called by the derived class when testing mocked packaged applications
+            protected Application() { }
```

Comment:
As mentioned in the comment, to create a derived class of the packaged application this constructor without arguments is needed. When an object of the derived class is created, **the constructor of the base class is called**. If  instead we create a constructor with multiple arguments, the base class constructor is called `Application(AppxPackageHelper.IAppxManifestApplication manifestApp, UWP package)` and that throws errors.

Reply:
I feel that we are adding functionalities just for the purpose of testing this class. IMO the correct way to test this class would be to create an object using `Application(AppxPackageHelper.IAppxManifestApplication manifestApp, UWP package)` constructor and test the functionalities. @ryanbodrug-microsoft can tell more about this.

Reply:
I added a null check in the constructor so that the code in the `Application(AppxPackageHelper.IAppxManifestApplication manifestApp, UWP package)` constructor does not throw any exception when either of the arguments is null. This way we won't need the extra constructor which does not have any arguments.